### PR TITLE
EZP-24859: Added a method to check the login availability

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-24859

# Description

needed to validate the login in the ezuser fields
Use https://github.com/ezsystems/ezpublish-kernel/pull/1415 which introduces the resources to do that.
See https://github.com/ezsystems/PlatformUIBundle/pull/345

# Tests

manual + unit tests

### CODE REVIEW

**Comment by dpobel** [comment]
ping @andrerom @mhyndle @StephaneDiot @yannickroger

**Comment by andrerom** [comment]
+1

**Comment by mhyndle** [comment]
+1

File: test/UserService.tests.js [hunk 1]
```diff
@@ -1009,6 +1015,62 @@ define(function (require) {
...
+                    userService.isLoginAvailable(login, mockCallback);
+                    expect(mockDiscoveryService.getInfoObject).toHaveBeenCalledWith("usersByLogin", jasmine.any(Function));
+                    expect(mockConnectionManager.request).toHaveBeenCalledWith(
+                        "HEAD", "/api/ezp/v2/user/users?login=" + login, '', {}, jasmine.any(Function)
+                    );
+                    mockConnectionManager.request.mostRecentCall.args[4].call(window, error, response);
+                    expect(mockCallback).toHaveBeenCalledWith(false, response);
+                });
+
+
```
Comment (yannickroger) [inline test/UserService.tests.js]: nitpick: extra line

**Comment by yannickroger** [comment]
+1

**Comment by dpobel** [comment]
waiting for https://github.com/ezsystems/ezpublish-kernel/pull/1415 to merge this one

**Pull request merged**